generator/src/main/java/org/corfudb/generator/LongevityApp.java

```diff
@@ -0,0 +1,168 @@
+    private boolean checkPoint;
+    BlockingQueue<Future> appsFutures;
+    BlockingQueue<Operation> operationQueue;
+    CorfuRuntime rt;
+    State state;
+
+    ExecutorService taskProducer;
+    ScheduledExecutorService checkpointer;
+    ExecutorService workers;
+
+    // How much time we live the application hangs once the duration is finished
+    // and the application is hanged
+    final int applicationTimeoutInMs = 20000;
+
+    final int queueCapacity = 1000;
+
+    long startTime;
+    int numberThreads;
+
+    public LongevityApp(long durationMs, int numberThreads, String configurationString, boolean checkPoint) {
+        this.durationMs = durationMs;
+        this.configurationString = configurationString;
+        this.checkPoint = checkPoint;
+        this.numberThreads = numberThreads;
+
+        operationQueue = new ArrayBlockingQueue<>(queueCapacity);
+        rt = new CorfuRuntime(configurationString).connect();
+        state = new State(50, 100, rt);
+
+        taskProducer = Executors.newSingleThreadExecutor();
+
+        workers = Executors.newFixedThreadPool(numberThreads);
+        checkpointer = Executors.newScheduledThreadPool(1);
+    }
+
+    /**
+     * Let the workers finish naturally (thanks to the timer) and then kill
+     * the producer and the checkpointer.
+     *
+     * At the end of the duration, we give some margin for the workers to finish their task before shutting
+     * down the thread pool.
+     * If the application is hung (which can happen when something goes wrong) we will still terminate.
+     */
+    private void waitForAppToFinish() {
+        workers.shutdown();
+        try {
+            workers.awaitTermination(durationMs + applicationTimeoutInMs, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        taskProducer.shutdownNow();
+
+        checkpointer.shutdownNow();
+    }
+
+    /**
+     * Check if we should still execute tasks or if the duration has elapsed.
+     *
+     * @return If we are still within the duration limit
+     */
+    private boolean withinDurationLimit() {
+        boolean result = System.currentTimeMillis() - startTime < durationMs;
+        return result;
+    }
+
+    /**
+     * A thread is tasked to indefinitely create operation. Operations are then
+     * put into a blocking queue. The blocking queue serves as a regulator for how
+     * many operations exist at the same time. It allows us to regulate the memory footprint.
+     *
+     * The only way to terminate the producer is to call explicitly a shutdownNow() on
+     * the executorService.
+     */
+    private void runTaskProducer() {
+        taskProducer.execute(() -> {
+            while (true) {
+                Operation current = state.getOperations().getRandomOperation();
+                try {
+                    operationQueue.put(current);
+                } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)